src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -610,15 +601,21 @@ public JsonPropertyInfo CreateJsonPropertyInfo(Type propertyType, string name)
                 ThrowHelper.ThrowArgumentException_CannotSerializeInvalidType(nameof(propertyType), propertyType, Type, name);
             }
 
-            JsonConverter converter = Options.GetConverterForType(propertyType);
-            JsonPropertyInfo propertyInfo = CreatePropertyUsingReflection(propertyType, converter);
+            if (!Options.IsInitializedForMetadataGeneration)
```

Comment:
As discussed, we should probably ensure this is done as soon as any JsonTypeInfo/JsonPropertyInfo is constructed with an options instance.